lib/app.dart

```diff
@@ -27,6 +27,24 @@ class MyApp extends StatelessWidget {
     return supportedLocales.first;
   }
 
+  // locale format: {languageCode}_{scriptCode}_{countryCode}
+  static Locale parseLocale(String locale) {
```

Comment:
Is there any built-in alternative?

Perhaps we should use the built-in parse function (if any) for better edge case handling.